src/serde_amount.rs

```diff
@@ -0,0 +1,32 @@
+//! Serde serialization for amount as msat
+
+use cln_rpc::primitives::Amount;
+
+pub trait SerdeAmount: Copy + Sized {
+    fn from_msat(msats: u64) -> Self;
+    fn as_msat(&self) -> u64;
+}
```

Comment:
Was there any particular reason for making the impl generic over this trait?

Reply:
No good reason, just what I got to work looking at `rust-bitcoin`

Reply:
Removed the trait in number #10. As its not necessary as we only need the one unit unlike `rust-bitcoin`? Do let me know if that's not what you meant

Reply:
Yep, that was exactly my point :)